deploy/eck-fleet-server/templates/fleet-server.yaml

```
@@ -0,0 +1,25 @@
+---
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: {{ include "fleet-server.fullname" . }}
+  labels:
+    {{- include "fleet-server.labels" . | nindent 4 }}
+  annotations:
+    eck.k8s.elastic.co/license: enterprise
+    {{- if .Values.annotations }}
+    {{- toYaml .Values.annotations | indent 4 }}
+    {{- end }}
+spec:
+  version: {{ required "A Fleet Server Agent version is required" .Values.version }}
+  {{- if not (hasKey .Values.spec "mode") }}
+  {{- fail "spec.mode of 'fleet' is required for fleet-server" }}
+  {{- else if (not (eq .Values.spec.mode "fleet")) }}
+  {{- fail "spec.mode of 'fleet' is required for fleet-server" }}
+  {{- end }}
+  {{- if not (hasKey .Values.spec "fleetServerEnabled") }}
+  {{- fail "spec.fleetServerEnabled is required to be true for fleet-server" }}
+  {{- else if (not (.Values.spec.fleetServerEnabled)) }}
+  {{- fail "spec.fleetServerEnabled is required to be true for fleet-server" }}
+  {{- end }}
```

Why do we allow the user to change these settings in deploy/eck-fleet-server/values.yaml if they have "fixed" and mandatory values?

```suggestion
  mode: "fleet"
  fleetServerEnabled: true
  {{- if (hasKey .Values.spec "mode") }}
  {{- fail "spec.mode cannot be changed" }}
  {{- end }}
  {{- if (hasKey .Values.spec "fleetServerEnabled") }}
  {{- fail "spec.fleetServerEnabled cannot be changed" }}
  {{- end }}
  {{- toYaml .Values.spec | nindent 2 }}
```

---

Makes sense, and updated.